src/domain/enrolment/editButtonPanel/EditButtonPanel.tsx

```diff
@@ -0,0 +1,135 @@
+  registration,
+}) => {
+  const { t } = useTranslation();
+  const authenticated = useSelector(authenticatedSelector);
+  const locale = useLocale();
+  const isMobile = useIsMobile();
+  const history = useHistory<EnrolmentsLocationState>();
+  const { search } = useLocation();
+
+  const goBack = () => {
+    const { returnPath, remainingQueryString } = extractLatestReturnPath(
+      search,
+      ROUTES.REGISTRATION_ENROLMENTS.replace(
+        ':registrationId',
+        registration.id as string
+      )
+    );
+
+    history.push({
+      pathname: `/${locale}${returnPath}`,
+      search: remainingQueryString,
+      state: { enrolmentId: enrolment.id },
+    });
+  };
+
+  const getActionItemProps = ({
+    action,
+    onClick,
+  }: {
+    action: ENROLMENT_EDIT_ACTIONS;
+    onClick: () => void;
+  }): MenuItemOptionProps | null => {
+    return getEditButtonProps({
+      action,
+      authenticated,
+      enrolment,
+      onClick,
+      t,
+    });
+  };
+
+  const actionItems: MenuItemOptionProps[] = [
+    getActionItemProps({
+      action: ENROLMENT_EDIT_ACTIONS.SEND_MESSAGE,
+      onClick: () => toast.error('TODO: Send message to attendee'),
+    }),
+    getActionItemProps({
+      action: ENROLMENT_EDIT_ACTIONS.CANCEL,
+      onClick: () => toast.error('TODO: Cancel enrolment'),
+    }),
+  ].filter((i) => i) as MenuItemOptionProps[];
+
+  return (
+    <div className={styles.editButtonPanel}>
+      <Container withOffset={true}>
+        <FormContainer>
+          <div className={styles.buttonsRow}>
+            <div className={styles.buttonWrapper}>
+              <Button
+                className={classNames(styles.backButton, styles.smallButton)}
+                iconLeft={<IconArrowLeft aria-hidden />}
+                fullWidth={true}
+                onClick={goBack}
+                type="button"
+                variant="secondary"
+              >
+                {t('common.buttonBack')}
+              </Button>
+              <div className={styles.actionsDropdown}>
+                <MenuDropdown
+                  button={
+                    isMobile ? (
+                      <button className={styles.toggleButton}>
+                        <IconMenuDots aria-hidden={true} />
+                      </button>
+                    ) : undefined
+                  }
+                  buttonLabel={t('common.buttonActions')}
+                  closeOnItemClick={true}
+                  items={actionItems}
+                  menuPosition="top"
+                />
+              </div>
+            </div>
+            <div className={styles.buttonColumn}>
+              <Button
+                className={styles.button}
+                fullWidth={true}
+                onClick={onSave}
+                type="button"
+              >
+                {t('enrolment.form.buttonSave')}
+              </Button>
+            </div>
+          </div>
+        </FormContainer>
+      </Container>
+    </div>
```

Comment:
`EditButtonPanel` is in three places and they seems to have many commonalities. Maybe this could be good time to see what parts are common and could there be reusable Panel component 🤔